pkg/lib/oauth/handler/handler_authz.go

```diff
@@ -109,19 +109,38 @@ func (h *AuthorizationHandler) doHandle(
 	}
 
 	s := session.GetSession(h.Context)
+
 	authnOptions := webapp.AuthenticateURLOptions{}
-	authnOptions.Prompt = r.Prompt()
 
-	// start web app authentication
-	if !slice.ContainsString(r.Prompt(), "none") {
-		// reset prompt to none for webapp to redirect back to authz endpoint
-		// after authentication
-		r2 := protocol.AuthorizationRequest{}
-		for k, v := range r {
-			r2[k] = v
+	// Handle max_age and prompt=login
+	prompt := r.Prompt()
+	if maxAge, ok := r.MaxAge(); ok {
+		impliesPromptLogin := false
+		// When there is no session, the presence of max_age implies prompt=login.
+		if s == nil {
```

Comment:
We should check `s == nil || s.SessionType() != session.TypeIdentityProvider` here. Although in the normal case authz endpoint should not be called with access token, we should only accept idp session here. We have the same checking when we handle prompt=none.